cognite/client/utils/_validation.py

```diff
@@ -8,6 +8,48 @@
 if TYPE_CHECKING:
     from cognite.client.utils._identifier import T_ID
 
+RESERVED_EXTERNAL_IDS = frozenset(
+    {
+        "Query",
+        "Mutation",
+        "Subscription",
+        "String",
+        "Int32",
+        "Int64",
+        "Int",
+        "Float32",
+        "Float64",
+        "Float",
+        "Timestamp",
+        "JSONObject",
+        "Date",
+        "Numeric",
+        "Boolean",
+        "PageInfo",
+        "File",
+        "Sequence",
+        "TimeSerie",
+    }
+)
+RESERVED_SPACE_IDS = frozenset({"space", "cdf", "dms", "pg3", "shared", "system", "node", "edge"})
+
+RESERVED_PROPERTIES = frozenset(
+    {
+        "space",
+        "externalId",
+        "createdTime",
+        "lastUpdatedTime",
+        "deletedTime",
+        "edge_id",
+        "node_id",
+        "project_id",
+        "property_group",
+        "seq",
+        "tg_table_name",
+        "extensions",
+    }
+)
```

Comment:
Having api-specific validation in generic validation modules adds to cognitive load, it's better to just keep it close to where it's used (e.g. `data_classes/data_modeling/_validation.py`)